R/setkey.R

```diff
@@ -356,8 +356,6 @@ CJ = function(..., sorted = TRUE, unique = FALSE)
       if (unique) l[[i]] = unique(y)
     }
   }
-  nrow = prod( vapply_1i(l, length) )  # lengths(l) will work from R 3.2.0
```

Comment:
migrated this to C, epsilon more efficient but anyway cleaner to have more logic there; will also make it easier to switch to long vector support